packages/core/js-client/src/util/libp2pUtils.ts

```diff
@@ -14,22 +14,26 @@
  * limitations under the License.
  */
 
-import { RelayOptions } from '@fluencelabs/interfaces';
-import { multiaddr, Multiaddr } from '@multiformats/multiaddr';
-import { isString } from './utils.js';
+import { RelayOptions } from "@fluencelabs/interfaces";
+import { multiaddr, Multiaddr } from "@multiformats/multiaddr";
+
+import { isString } from "./utils.js";
 
 export function relayOptionToMultiaddr(relay: RelayOptions): Multiaddr {
-    const multiaddrString = isString(relay) ? relay : relay.multiaddr;
-    const ma = multiaddr(multiaddrString);
+  const multiaddrString = isString(relay) ? relay : relay.multiaddr;
+  const ma = multiaddr(multiaddrString);
+
+  const peerId = ma.getPeerId();
 
-    throwIfHasNoPeerId(ma);
+  if (peerId == null) {
+    throwHasNoPeerId(ma);
+  }
```

Comment:
I would use this function directly in `RelayConnection.ts` and remove `throwHasNoPeerId` completely. So that peerId existance is always checked and there is no way to forget checking for that

Reply:
This function is used in 2 places - 
`RelayConnections.ts`
and
at `ClientPeer.ts` via `relayOptionToMultiaddr`

Reply:
I know `throwHasNoPeerId` is used in two places that's exactly why I suggest the change. I suggest to not use `multiaddr` function directly but use this `relayOptionToMultiaddr` function in `RelayConnections.ts` because logic of getting multiaddr checking it has peerid and throwing if it doesn't is completely duplicated

Reply:
I don't understand what you're proposing.
relayOptionToMultiaddr operates on `RelayOptions` type
and
RelayConnection `constructor`operates on `RelayConnectionConfig` type

It's different types which can contain different types of multiaddr(for example in RelayOptions it's a plain string) and processing these configs is different.

Reply:
I went through the chain of classes just now. Is the getPeerId check needed at all in `RelayConnection.ts`? Seems like multiaddr that is checked there - we already checked it when `relayOptionToMultiaddr` was used to generate this multiaddr in the first place? Can this multiaddr ever not contain peerId somehow? If it can and this check is required - please feel free to resolve the issues. I just don't like duplicated checkes and a possibiblity to have invalid multiaddr object flying around the codebase for some reason. My reasoning is that it would be good to always produce mutltiaddr objects that are validated right away when they are first created. That seems more secure and simple then validating them again arbitrarely for some reason

Reply:
Yes, you're right. These check are dublicates without any specific reason. We can validate config once at creation stage and use validated config later. But this change will take a bit more refactoring than this - need to change class definitions and logic of config processing. I will do this in other task related to input validation.